### PR TITLE
Tile op implementation

### PR DESCRIPTION
**Pull request by jianyinglang**

### CODE REVIEW

File: src/ngraph_builder.cc [hunk 1]
```diff
@@ -2326,6 +2326,72 @@ tf::Status Builder::TranslateGraph(const std::vector<tf::TensorShape>& inputs,
...
+      }
+      std::vector<tf::int64> multiples;
+      TF_RETURN_IF_ERROR(tf::GetNodeAttr(
+          op->attrs(), "_ngraph_tile_static_multiples", &multiples));
+      auto ng_input_shape = ng_input->get_shape();
+      if (ng_input_shape.size() != multiples.size()) {
+        return tf::errors::InvalidArgument(
+            "dimension of input does not match length of multiples");
+      }
+      std::shared_ptr<ng::Node> ng_output = ng_input;
```
Comment (aprocter) [inline src/ngraph_builder.cc]: No need to initialize this to `ng_input`.

File: test/python/test_tile.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                 z: z_np
+             })
+
+    print("result_a:", result_a)
+    print("expected_a:", expected_a)
+    print("result_a:", result_b)
+    print("expected_a:", expected_b)
+    atol = 1e-5 
+    error_a = np.absolute(result_a-expected_a)
+    #error_b = np.absolute(result_b-expected_b)
```
Comment (aprocter) [inline test/python/test_tile.py]: Let's remove the commented-out code here.

File: test/python/test_tile.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    assert np.amax(error_a) <= atol
+    #assert np.amax(error_b) <= atol
```
Comment (aprocter) [inline test/python/test_tile.py]: Let's remove the commented-out code here.

File: test/tf_exec.cpp [hunk 4]
```diff
@@ -278,6 +278,38 @@ TEST(tf_exec, BatchMatMul_2D) {
...
+  for (int i = 0; i < A_flat.size(); i++) {
+    A_flat.data()[i] = -1.1f*i;
+  }
+  auto X = tf::ops::Const(root, {tf::int64(3),  tf::int64(4), tf::int64(2)}, tf::TensorShape({3})); 
+  auto Y = tf::ops::Const(root, {tf::int64(1),  tf::int64(0), tf::int64(3)}, tf::TensorShape({3}));
+  auto C = tf::ops::Tile(dev_scope.WithOpName("C"), A, X);
+  auto D = tf::ops::Tile(dev_scope.WithOpName("D"), A, Y);
+  std::vector<tf::Tensor> outputs_C;
+  std::vector<tf::Tensor> outputs_D;
+  // Run and fetch R
```
Comment (aprocter) [inline test/tf_exec.cpp]: Is this comment valid?

File: test/tf_exec.cpp [hunk 5]
```diff
@@ -278,6 +278,38 @@ TEST(tf_exec, BatchMatMul_2D) {
...
+  tf::ClientSession session(dev_scope);
+  TF_CHECK_OK(session.Run({C}, &outputs_C));
+  TF_CHECK_OK(session.Run({D}, &outputs_D));
+  // Expect outputs[0] == [19; -3]
```
Comment (aprocter) [inline test/tf_exec.cpp]: Is this comment valid?

File: src/ngraph_builder.cc (hunk 1 above)
Comment (aprocter) [reply src/ngraph_builder.cc]: Er, wait, yes there is  :)

File: test/tf_exec.cpp (hunk 4 above)
Comment (jianyinglang) [reply test/tf_exec.cpp]: Removed the comment

File: test/tf_exec.cpp (hunk 5 above)
Comment (jianyinglang) [reply test/tf_exec.cpp]: Removed

File: src/ngraph_builder.cc [hunk 6]
```diff
@@ -2258,6 +2258,72 @@ tf::Status Builder::TranslateGraph(const std::vector<tf::TensorShape>& inputs,
...
+        for (int i=0; i<ng_input_shape.size(); i++) {
+          if (multiples[i] < 0) {
+            return tf::errors::InvalidArgument("Expected multiples[", i, "] >= 0, but got ",
+                                  multiples[i]); 
+          }
+          vector<shared_ptr<ng::Node>> tmp_tensors;
+          for (int k=0; k<multiples[i]; k++) {
+            tmp_tensors.push_back(ng_output);
+          }
+          auto ng_concat = make_shared<ngraph::op::Concat>(tmp_tensors, i);
```
Comment (aprocter) [inline src/ngraph_builder.cc]: One more thing if you get a chance: I think we can skip here if `multiples[i]==1`, right?

**Pull request merged**